### PR TITLE
Install Istio and Gloo-mesh with Istio Test Framework

### PR DESCRIPTION
**Pull request by nmnellis**
Using the test framework you can can easily and quickly deploy Istio/Gloo-mesh to many existing clusters and run a full suite of integration tests between them. This current implementation installs Istio and Gloo-mesh as well as a suite of applications that can be used for testing different routing situations. They are installed on every cluster can can be reconfigured based on each test case.

Had to update Istio to 1.9.4 which made me have to upgrade k8s libs. that was not fun

BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh/issues/1507

### CODE REVIEW

File: pkg/meshctl/commands/install/install.go [hunk 1]
```diff
@@ -214,70 +215,73 @@ func enterpriseCommand(ctx context.Context, installOpts *options) *cobra.Command
...
 
-func (o enterpriseOptions) getInstaller() helm.Installer {
-	ins := o.options.getInstaller(gloomesh.GlooMeshEnterpriseChartUriTemplate)
-	ins.ReleaseName = o.releaseName
-	ins.Values["licenseKey"] = o.licenseKey
-	if o.skipUI {
+func (o EnterpriseOptions) getInstaller() helm.Installer {
+	ins := o.Options.getInstaller(gloomesh.GlooMeshEnterpriseChartUriTemplate)
+	ins.ReleaseName = o.ReleaseName
+	ins.Values["LicenseKey"] = o.LicenseKey
```
Comment (joekelley) [inline pkg/meshctl/commands/install/install.go:246]:
```suggestion
	ins.Values["licenseKey"] = o.LicenseKey
```

File: pkg/test/apps/echo/certs/gen-certs.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:4096 -subj "/C=US/ST=MA/O=Solo.io" -keyout echo-ca.key -out echo-ca.crt
```
Comment (joekelley) [inline pkg/test/apps/echo/certs/gen-certs.sh:3]: curious, what version of openssl are you using here? We've had issues with documenting openssl commands because macos ships with LibreSSL which doesn't support all the same flags as openssl

File: pkg/test/apps/gloo-mesh/gloomesh.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+type Config struct {
+	ClusterKubeConfigs                  map[string]string
+	DeployControlPlaneToManagementPlane bool
+}
+
+const glooMeshVersion = "1.1.0-beta8"
```
Comment (joekelley) [inline pkg/test/apps/gloo-mesh/gloomesh.go:17]: could this be read from an env var?

File: pkg/test/apps/gloo-mesh/kube.go [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+		}
+	} else {
+		if err := i.deployControlPlane(licenceKey); err != nil {
+			return nil, err
+		}
+	}
+	return i, err
+}
+
+func (i *glooMeshInstance) deployManagementPlane(licenceKey string) error {
```
Comment (joekelley) [inline pkg/test/apps/gloo-mesh/kube.go:62]: nit: licenceKey => licenseKey

File: pkg/meshctl/install/helm/installer.go [hunk 5]
```diff
@@ -86,7 +88,7 @@ func (i Installer) InstallChart(ctx context.Context) error {
 		valueOpts.ValueFiles = []string{valuesFile}
 	}
 	for key, value := range i.Values {
-		valueOpts.Values = append(valueOpts.Values, key+"="+value)
+		valueOpts.Values = append(valueOpts.Values, lowerFirst(key)+"="+value)
```
Comment (joekelley) [inline pkg/meshctl/install/helm/installer.go:91]: is this change necessary? upper case keys are not recommended but technically supported by helm

File: pkg/meshctl/registration/register.go [hunk 6]
```diff
@@ -29,6 +29,7 @@ var gloomeshRbacRequirements = func() []rbacv1.PolicyRule {
 
 type Options struct {
 	KubeConfigPath         string
+	MgmtKubeConfigPath     string
```
Comment (joekelley) [inline pkg/meshctl/registration/register.go:32]: @rvennam FYI -- this should resolve https://github.com/solo-io/gloo-mesh/issues/1547

**Review by joekelley — COMMENTED** [review]
changes affecting production codepaths look good, just some nits

File: pkg/test/apps/echo/certs/gen-certs.sh (hunk 2 above)
Comment (nmnellis) [reply pkg/test/apps/echo/certs/gen-certs.sh:3]: ▶ openssl version
LibreSSL 2.8.3

File: pkg/meshctl/install/helm/installer.go (hunk 5 above)
Comment (nmnellis) [reply pkg/meshctl/install/helm/installer.go:91]: nice find, this was due to the above problem where i accidentally changed licenseKey to LicenseKey. i can clean this up

File: pkg/meshctl/registration/register.go (hunk 6 above)
Comment (nmnellis) [reply pkg/meshctl/registration/register.go:32]: btw i did no expose this to the cli. if you would like me to i can

File: pkg/test/apps/gloo-mesh/gloomesh.go (hunk 3 above)
Comment (nmnellis) [reply pkg/test/apps/gloo-mesh/gloomesh.go:17]: yes

File: pkg/meshctl/registration/register.go (hunk 6 above)
Comment (joekelley) [reply pkg/meshctl/registration/register.go:32]: yes, please do it'll be helpful to have

Comment (nmnellis) [reply pkg/meshctl/registration/register.go:32]: its been added https://github.com/solo-io/gloo-mesh/pull/1503/files#diff-34af8719c98f18ed4329f3b7f304c5250e0bd25e8b39db7b37e630ddb8d03aebR31

File: pkg/test/apps/gloo-mesh/gloomesh.go (hunk 3 above)
Comment (nmnellis) [reply pkg/test/apps/gloo-mesh/gloomesh.go:18]: https://github.com/solo-io/gloo-mesh/pull/1503/files#diff-6b63ca90085c7ee502f75618339d996adf08fb7833ace921c44c11f5242f69d0R29

**Review by joekelley — APPROVED** [review]
